File: Top_Interview_150/Array_String/python/55.py
from typing import List


class Solution:
    def canJump(self, nums: List[int]) -> bool:
        n = len(nums)
        max_index = 0

        for i in range(n - 1):
            if i <= max_index:
                max_index = max(max_index, i + nums[i])

        return n - 1 <= max_index

File: Top_Interview_150/Array_String/cpp/55.cpp
#include <vector>
#include <algorithm>

using namespace std;

class Solution
{
public:
    bool canJump(vector<int> &nums)
    {
        int n = nums.size();
        int max_index = 0;

        for (int i = 0; i < n - 1; ++i)
        {
            if (i <= max_index)
                max_index = max(max_index, i + nums[i]);
        }

        return n - 1 <= max_index;
    }
};